pallets/identity/src/benchmarking.rs

```diff
@@ -295,6 +305,50 @@ benchmarks! {
         Module::<T>::unsafe_join_identity(target.did(), Permissions::empty(), key.account());
     }: _(target.origin, signatory, Permissions::default().into())
 
+    permissions_cost {
+        // Number of assets/portfolios/pallets/extrinsics.
+        let a in 0 .. MAX_ASSETS; // a=(A)ssets
+        let p in 0 .. MAX_PORTFOLIOS; // p=(P)ortfolios
+        let l in 0 .. MAX_PALLETS; // l=pa(L)lets
+        let e in 0 .. MAX_EXTRINSICS; // e=(E)xtrinsics
+
+        let asset = AssetPermissions::elems(
+            (0..a as u64).map(Ticker::generate_into)
+        );
+        let portfolio = PortfolioPermissions::elems(
+            (0..p as u128).map(|did| {
+                PortfolioId::user_portfolio(did.into(), PortfolioNumber(0))
+            })
+        );
+        let dispatchable_names = DispatchableNames::elems(
+            (0..e as u64).map(|e| {
+                DispatchableName(Ticker::generate(e))
+            })
+        );
+        let extrinsic = ExtrinsicPermissions::elems(
+            (0..l as u64).map(|p| {
+                PalletPermissions {
+                    pallet_name: PalletName(Ticker::generate(p)),
+                    dispatchable_names: dispatchable_names.clone(),
+                }
+            })
+        );
+
+        let permissions = Permissions {
+            asset,
+            extrinsic,
+            portfolio
+        };
+
+        let orig = permissions.clone();
+    }: {
+        let enc = permissions.encode();
+        // Panic the benchmarks if decoding fails.
+        let perm = Permissions::decode(&mut enc.as_slice())
+            .expect("Permissions should decode correctly.");
+        assert_eq!(orig, perm);
```

Comment:
I wanted this to do some work based on how complex the permissions are.  Also wanted to make sure the compiler doesn't optimize way all the work.

The pallet/extrinsic permissions are the most complex, because each pallet/extrinsic name is one heap allocation + heap allocation for the BTreeSet.

Reply:
> Also wanted to make sure the compiler doesn't optimize way all the work.

Should still happen if you move `.expect` + `assert_eq` to valdiation.

Reply:
I still want that work in the benchmark.

Reply:
Not sure why, but alright.